manual/src/refman/extensions/externaltypes.etex

```diff
@@ -0,0 +1,54 @@
+(Introduced in OCaml 5.4)
+
+External types introduce types of values that are defined outside the
+OCaml language (they cannot be defined as standard datatype or formed
+by combining existing type constructors) because they are populated
+from the foreign function interface. For example, a type "gmp" of
+arbitrary-length integers implementing by binding the "Gmp" library
+from C could be defined as follows:
+
+\begin{caml_example*}{verbatim}
+module Gmp :
+  type t = external "gmp"
+
+  val of_int : int -> t
+  val add : t -> t -> t
+  (* ... *)
+end
+\end{caml_example*}
+
+We say that "Gmp.t" is an external type of name
+\machine{"gmp"}. External names are not unique, it is possible to
+define two (distinct) external types with the same name. On the other
+hand, the type-system enforces that two external types with distinct
+names are different.
+
+Before external types were available, users could use an abstract type
+for this purpose, just with "type gmp". But this is less informative,
+as the OCaml compiler must assume that abstract types could be defined
+in an arbitrary way, and in particular an abstract type from another
+module could be equal to any type.
+
+\paragraph{Predefined types}
+
+External types are used in particular to define the built-in types
+such as "int", "bool", etc.
+
+\begin{caml_example*}{verbatim}
+type int = external "int" [@@immediate]
+type bool = external "bool" [@@immediate]
```

Comment:
The `immediate` attribute is not supported on external types currently. I would rather avoid mentioning features that don't exist in the documentation.

Reply:
Hm, isn't it an issue that this is not supported? If declaring builtin types is the main use-cases of external types, I would expect that shape/representation annotations are supported.